File: app/app/src/main/java/me/williamhester/reddit/BreaditApplication.java
package me.williamhester.reddit;

import android.app.Application;

import me.williamhester.reddit.inject.ApplicationComponent;
import me.williamhester.reddit.inject.DaggerApplicationComponent;

/** Contains the Dagger component for the application. */
public class BreaditApplication extends Application {

  private ApplicationComponent applicationComponent;

  @Override
  public void onCreate() {
    super.onCreate();

    applicationComponent = DaggerApplicationComponent.builder().build();
  }

  public ApplicationComponent getApplicationComponent() {
    return applicationComponent;
  }
}

File: app/app/src/main/java/me/williamhester/reddit/BreaditApplication.kt
package me.williamhester.reddit

import android.app.Application
import io.realm.Realm

import me.williamhester.reddit.inject.ApplicationComponent
import me.williamhester.reddit.inject.ApplicationModule
import me.williamhester.reddit.inject.DaggerApplicationComponent

/** Contains the Dagger component for the application.  */
class BreaditApplication : Application() {

  lateinit var applicationComponent: ApplicationComponent
    private set

  override fun onCreate() {
    super.onCreate()

    Realm.init(this)
    applicationComponent =
        DaggerApplicationComponent
            .builder()
            .applicationModule(ApplicationModule(this))
            .build()
  }
}
